src/app/features/item/item-template/item-preview.service.ts

```diff
@@ -0,0 +1,1349 @@
+    }
+
+    return gemEnchantmentText;
+  }
+
+  public async getSocketEnchantment(): Promise<string> {
+
+    let socketText = '';
+
+    // fill native sockets
+    for (let j = 1; j <= 3; j++) {
+      const socketColor = this.editorService.form.controls['socketColor_' + j].value;
+
+      if (!socketColor) {
+        continue;
+      }
+
+      let colorId = 0;
+      for (let i = 0; i < 4; i++) {
+        if (socketColor & (1 << i)) {
+          colorId = i;
+        }
+      }
+
+      socketText += `<br><span class="socket-${ITEM_CONSTANTS.gemColors[colorId]} q0 socket${colorId}">${ITEM_CONSTANTS.socket[colorId]}</span>`;
+    }
+
+    const socketBonus = this.editorService.form.controls.socketBonus.value;
+    if (!!socketBonus) {
+      const sbonus = await this.sqliteQueryService.getSocketBonusById(socketBonus);
+      const socketBonusText = `${ITEM_CONSTANTS.socketBonus.replace('%s', sbonus)}`;
+      socketText += `<br><span class="q0">${socketBonusText}</span>`;
+    }
+
+    return socketText;
+  }
+
+  public async getSpellDesc(green: string[]) {
+    const spellId1 = this.editorService.form.controls.spellid_1.value;
+    const spellId2 = this.editorService.form.controls.spellid_2.value;
+
+    if (!this.canTeachSpell(spellId1, spellId2)) {
+      const itemSpellsAndTrigger = [];
+      for (let j = 1; j <= 5; j++) {
+        const spellid = this.editorService.form.controls['spellid_' + j].value;
+
+        if (spellid > 0) {
+          let cooldown = this.editorService.form.controls['spellcooldown_' + j].value;
+          const cooldownCategory = this.editorService.form.controls['spellcategory_' + j].value;
+
+          if (cooldown < cooldownCategory) {
+            cooldown = cooldownCategory;
+          }
+
+          cooldown = cooldown < 5000 ? '' : ` ( ${this.formatTime(cooldown)} cooldown)`;
+
+          itemSpellsAndTrigger[spellid] = [this.editorService.form.controls['spelltrigger_' + j].value, cooldown];
+        }
+      }
+
+      if (itemSpellsAndTrigger) {
+        const spellIDs = Object.keys(itemSpellsAndTrigger);
+        for (const spellID of spellIDs) {
+          const spellTrigger = itemSpellsAndTrigger[spellID];
+          const parsed = await this.sqliteQueryService.getSpellDescriptionById(spellID); // TODO: parseText correctly
+
+          green.push(ITEM_CONSTANTS.trigger[spellTrigger[0]] + parsed + spellTrigger[1]);
+        }
+
+      }
+    }
+  }
+
+  // TODO: recipes, vanity pets, mounts
+  public async getLearnSpellText(): Promise<string> {
+    /* TODO - WIP */
+
+    let spellDesc = '';
+
+    // const bagFamily: number = Number(this.editorService.form.controls.BagFamily.value);
+    // const itemClass: number = Number(this.editorService.form.controls.class.value);
+    const spellId1 = this.editorService.form.controls.spellid_1.value;
+    const spellId2 = this.editorService.form.controls.spellid_2.value;
+
+    if (this.canTeachSpell(spellId1, spellId2)) {
+      const craftSpell = spellId2;
```

Comment:
Local variable 'craftSpell' is redundant

Reply:
this is a WIP part, I will not touch this until this https://github.com/azerothcore/Keira3/issues/527